src/net.h

```diff
@@ -144,8 +144,8 @@ enum
 };
 
 bool IsPeerAddrLocalGood(CNode *pnode);
-/** Returns a local address that we should advertise to this peer */
-std::optional<CAddress> GetLocalAddrForPeer(CNode *pnode);
+/** Returns a local address that we should advertise to this peer. */
+std::optional<CService> GetLocalAddrForPeer(CNode* pnode);
```

Comment:
4c59e37c18b177fc50786de37ffffb52de6daaa7: While touching this method, could switch `*` to `&`? (And remove "p")